src/io/limits.rs

```diff
@@ -0,0 +1,15 @@
+/// Limits on how much memory that can be used in the decoding process. These limits are "best
+/// effort" limits and there are no strict guarantees that they will be followed at all times.
+#[derive(Debug, Copy, Clone)]
+pub struct DecodingLimits {
+    /// A limit on how large buffers a decoder is allowed to allocate. The default is 512MiB.
+    pub buffer_limit: usize,
```

Comment:
Some formats already demonstrated the potential usefulness of more fine-grained limits. There should be some plan on extending the limit system, so this should not be a completely public struct.